### PR TITLE
[CH] Fix inconsistent results of position function for ch backend

### PR DESCRIPTION
**Pull request by exmy**
## What changes were proposed in this pull request?

 Fix inconsistent results when the start_pos parameter of the position function is null

### CODE REVIEW

**Comment by exmy** [comment]
@zzcclp please take a review

File: backends-clickhouse/src/test/scala/io/glutenproject/execution/GlutenClickHouseTPCHParquetSuite.scala [hunk 1]
```diff
@@ -658,17 +658,20 @@ class GlutenClickHouseTPCHParquetSuite extends GlutenClickHouseTPCHAbstractSuite
...
         |position('I', l_shipinstruct, 5), position('IN', l_shipinstruct),
         |position('', l_shipinstruct), locate(l_returnflag, l_shipinstruct),
         |position(l_returnflag in l_shipinstruct), position('bar', 'foobarbar'),
         |position(l_returnflag, 'TENSTNTEST', 4), position('bar', 'foobarbar', 5),
-        |position(l_returnflag, l_shipinstruct, l_linenumber + 11)
-        |from lineitem where l_orderkey = 1 and l_partkey = 15519
+        |position(l_returnflag, l_shipinstruct, l_linenumber + 11),
+        |position(null, l_shipinstruct),
+        |position(l_returnflag, null),
+        |position(l_returnflag, l_shipinstruct, null),
```
Comment (zzcclp) [inline backends-clickhouse/src/test/scala/io/glutenproject/execution/GlutenClickHouseTPCHParquetSuite.scala:670]: add a case: `position(l_returnflag, l_shipinstruct, 0),`

File: gluten-core/src/main/scala/io/glutenproject/expression/TernaryExpressionTransformer.scala [hunk 2]
```diff
@@ -51,6 +55,53 @@ case class StringTranslateTransformer(substraitExprName: String, srcExpr: Expres
...
+      GlutenConfig.GLUTEN_CLICKHOUSE_BACKEND)) {
+
+      val substrNode = substrExpr.doTransform(args)
+      val strNode = strExpr.doTransform(args)
+      val startNode = startExpr.doTransform(args)
+
+      // Special Case
+      // In Spark, return 0 when start_pos is null
+      // but when start_pos is not null, return null if either str or substr is null
+      // so we need convert it to if(isnull(start_pos), 0, position(substr, str, start_pos)
```
Comment (zzcclp) [inline gluten-core/src/main/scala/io/glutenproject/expression/TernaryExpressionTransformer.scala:76]: if start_pos is 0, CH backend return 0 or null?

**Comment by zzcclp** [comment]
In Vanilla Spark, when start_pos is null or less than 1 , it will return 0, otherwise return null. Please check that.

File: backends-clickhouse/src/test/scala/io/glutenproject/execution/GlutenClickHouseTPCHParquetSuite.scala (hunk 1 above)
Comment (exmy) [reply backends-clickhouse/src/test/scala/io/glutenproject/execution/GlutenClickHouseTPCHParquetSuite.scala:670]: Done

File: gluten-core/src/main/scala/io/glutenproject/expression/TernaryExpressionTransformer.scala (hunk 2 above)
Comment (exmy) [reply gluten-core/src/main/scala/io/glutenproject/expression/TernaryExpressionTransformer.scala:76]: If start_pos is 0 and both str and substr are not null, return 0. 
However, if start_pos is 0 and either str or substr is null,  return null, which is consistent with the behavior of Spark.

**Review by zzcclp — APPROVED** [review]
LGTM

**Review by rui-mo — APPROVED** [review]

**Review by lviiii — APPROVED** [review]

**Pull request merged**